public/app/plugins/datasource/grafana/components/QueryEditor.tsx

```diff
@@ -65,6 +63,13 @@ export class QueryEditor extends PureComponent<Props, State> {
   }
 
   componentDidMount() {
+    const { query } = this.props;
+
+    // If there is no query type, set the default
+    if (!query.queryType) {
+      const defaultQueryType = this.queryTypes.find((v) => v.value === defaultQuery.queryType) || this.queryTypes[0];
+      this.onQueryTypeChange(defaultQueryType);
+    }
```

Comment:
We should not call model change callbacks on mount, is there a way to avoid this? (calling this triggers a bunch of events, and model persistence changes that should really be due to user action, not just mount)

Reply:
@torkelo guess it depends if you consider a query with just `{refId: 'A'}` in the json valid? if so, we can just not call anything on mount and it will all work. https://github.com/grafana/grafana/pull/37965/files#diff-1c565d0c1df80b370a946c7d533c0ea5a534593b5db63202e2c15075a4a51465R262 ensures if there's no matching `queryType` we default to the first. and as soon as you change the `queryType`, it's updated via `onQueryTypeChange`. that feels a little weird though because then there's a disconnect between the json model and what's displayed on screen.

Reply:
we need a way to set query defaults, not sure where that should be though.

Reply:
hmmm i see

`QueryGroup` has access to the datasource, so we could put an optional `defaultQuery` in the plugin.json file under `queryOptions` i guess.

i'm not sure what changes would be required to support that in the backend though, or if that even makes sense. i guess you'd then have to include `"refId": "A"` in that default query, which seems redundant. and you wouldn't be able to use any of the defined types. e.g. for the --Grafana-- datasource, there's an [enum for the query types](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/grafana/types.ts#L8) (and a [default query](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/grafana/types.ts#L21)) but if we put that in json we'd just have to know the internal strings and wouldn't be able to use those 😕 

i think i'm talking myself out of that already... 🤔